src/injected/single-frame-tab-stop-listener.ts

```diff
@@ -4,24 +4,22 @@
 import { DateProvider } from 'common/date-provider';
 import { TabStopEvent } from 'common/types/store-data/tab-stop-event';
 import { AllFrameRunnerTarget } from 'injected/all-frame-runner';
+import { ShadowDomFocusTracker } from 'injected/shadow-dom-focus-tracker';
 
-export class SingleFrameTabStopListener implements AllFrameRunnerTarget<TabStopEvent> {
+export class SingleFrameTabStopListener
+    extends ShadowDomFocusTracker
```

Comment:
Is there a particular reason we're using inheritance for this? Do we foresee this being a base class that others would use?

We generally favor using composition over inheritance unless there's a strong reason to use inheritance.

Reply:
@waabid I'm going to go ahead and merge this since it's been approved and has been sitting for so long, but let me know if you have any further feedback and I'll be happy to make a follow up PR to resolve it!